WebUI/src/lib/components/WiFiList.svelte

```diff
@@ -1,57 +1,58 @@
 <script lang="ts">
-	import { getModalStore } from '@skeletonlabs/skeleton';
-	import WiFiInfo from '$lib/components/modals/WiFiInfo.svelte';
-	import type { WiFiNetwork } from '$lib/types/WiFiNetwork';
-	import { WiFiStateStore } from '$lib/stores';
-	import { WebSocketClient } from '$lib/WebSocketClient';
+  import { getModalStore } from '@skeletonlabs/skeleton';
+  import WiFiInfo from '$lib/components/modals/WiFiDetails.svelte';
+  import type { WiFiNetwork } from '$lib/types/WiFiNetwork';
+  import { WiFiStateStore } from '$lib/stores';
+  import { WebSocketClient } from '$lib/WebSocketClient';
 
   const modalStore = getModalStore();
 
-  let connectedIndex = -1;
+  let connectedBSSID: string | null = null;
 
   function wifiScan() {
-    WebSocketClient.Instance.Send('{ "type": "startScan" }');
+    if ($WiFiStateStore.scanning) {
+      WebSocketClient.Instance.Send('{ "type": "wifi", "action": "scan", "run": false }');
+    } else {
+      WebSocketClient.Instance.Send('{ "type": "wifi", "action": "scan", "run": true }');
+    }
```

Comment:
I think it's good that you defined the protocol in a separate file `WebSocketMessageHandler`, but you also defined the handlers there, yet nowhere did you define the _transmission_ methods other than raw JSON strings here.

Maybe something like this:
- `WebSocketProtocol.ts`
- `WebSocketMessageReceiver.ts`
- `WebSocketMessageSender.ts`
..would make more sense in terms of structure?

And then you can:
```ts
import { sendWifiScan } from 'WebSocketMessageSender';
// ...
sendWifiScan(WebSocketClient.Instance)
```

This way you can keep all the protocol stuff in one place, and handle the actual logic surrounding it in another.

Reply:
Possibly... But json wont be around for that long so doesnt matter